### PR TITLE
Implementing `list` and `list_permissions` Commands.

### PR DESCRIPTION
**Pull request by bdchatham**
Implementing `list` and `list_permissions` Commands.

### What Changed?
 - Minor refactors for `can-call` command.
 - Added `user list` command.
 - Added `user list-permissions` command.
 - Added unit tests for both commands.

### CODE REVIEW

File: pkg/user/appointee/can_call.go [hunk 1]
```diff
@@ -66,8 +59,9 @@ func canCall(cliCtx *cli.Context, generator func(logging.Logger, *canCallConfig)
 	}
 
 	result, err := elReader.UserCanCall(ctx, config.UserAddress, config.CallerAddress, config.Target, config.Selector)
-	logger.Infof("CanCall Result: %v", result)
-	logger.Infof("Selector, Target and User: %s, %x, %s", config.Target, string(config.Selector[:]), config.UserAddress)
+	fmt.Printf("CanCall Result: %v", result)
```
Comment (shrimalmadhur) [inline pkg/user/appointee/can_call.go:62]: you need a `\n` for new line at the end.

File: pkg/user/appointee/can_call.go [hunk 2]
```diff
@@ -66,8 +59,9 @@ func canCall(cliCtx *cli.Context, generator func(logging.Logger, *canCallConfig)
...
+	fmt.Println()
+	fmt.Printf("Selector, Target and User: %s, %x, %s", config.Target, string(config.Selector[:]), config.UserAddress)
```
Comment (shrimalmadhur) [inline pkg/user/appointee/can_call.go:64]: you need a `\n` for new line at the end.

File: pkg/user/appointee/list.go [hunk 3]
```diff
@@ -1,25 +1,158 @@
...
 	"github.com/Layr-Labs/eigenlayer-cli/pkg/internal/common/flags"
 	"github.com/Layr-Labs/eigenlayer-cli/pkg/telemetry"
+	"github.com/Layr-Labs/eigenlayer-cli/pkg/utils"
+	"github.com/Layr-Labs/eigensdk-go/chainio/clients/elcontracts"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	eigenSdkUtils "github.com/Layr-Labs/eigensdk-go/utils"
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
 	"github.com/urfave/cli/v2"
+	"sort"
```
Comment (shrimalmadhur) [inline pkg/user/appointee/list.go:16]: `make fmt`

File: pkg/user/appointee/list.go [hunk 4]
```diff
@@ -1,25 +1,158 @@
...
+}
+
+func printResults(config *listUsersConfig, users []gethcommon.Address) {
+	fmt.Printf(
+		"Selector, Target and Appointer: %s, %x, %s",
+		config.Target,
+		string(config.Selector[:]),
+		config.UserAddress,
+	)
+	fmt.Println("====================================================================================")
```
Comment (shrimalmadhur) [inline pkg/user/appointee/list.go:78]: we can do `strings.Repeat` and specify a number. easy to read.

File: pkg/user/appointee/list.go [hunk 5]
```diff
@@ -1,25 +1,158 @@
...
+	generator func(logging.Logger, *listUsersConfig) (ListUsersReader, error),
+) error {
+	ctx := cliCtx.Context
+	logger := common.GetLogger(cliCtx)
+
+	config, err := readAndValidateListUsersConfig(cliCtx, logger)
+	if err != nil {
+		return eigenSdkUtils.WrapError("failed to read and validate user can call config", err)
+	}
+
```
Comment (shrimalmadhur) [inline pkg/user/appointee/list.go:57]: don't we have to set ctx metadata here for network?
`cCtx.App.Metadata["network"] = config.chainID.String()` 
I know it's not ideal. we need something better to have it applied automatically but we can add in all commands right now

**Comment by shrimalmadhur** [comment]
you probably need to rebase and fix the conflicts. unit tests should pass now as we fixed it in the target branch. IT will fail until my eigensdk PR gets merged.

File: pkg/user/appointee/can_call.go (hunk 1 above)
Comment (bdchatham) [reply pkg/user/appointee/can_call.go:62]: Had seen other places explicitly using println for new lines -- I'll update.

File: pkg/user/appointee/list.go (hunk 5 above)
Comment (bdchatham) [reply pkg/user/appointee/list.go:57]: I'm not clear on why that's needed. Can you explain?

**Comment by bdchatham** [comment]
@shrimalmadhur fmt, unit tests and lint are all passing. Unclear on the comments related to those.

File: pkg/user/appointee/list.go (hunk 5 above)
Comment (shrimalmadhur) [reply pkg/user/appointee/list.go:57]: https://github.com/Layr-Labs/eigenlayer-cli/blob/master/pkg/telemetry/telemetry.go#L59

**Review by shrimalmadhur — APPROVED** [review]

**Pull request merged**